### PR TITLE
Add option to run only specific sub scan

### PR DESCRIPTION
**Pull request by attiasas**
- [x] The pull request is targeting the `dev` branch.
- [x] The code has been validated to compile successfully by running `go vet ./...`.
- [x] The code has been formatted properly using `go fmt ./...`.
- [x] All [static analysis checks](https://github.com/jfrog/jfrog-cli-security/actions/workflows/analysis.yml) passed.
- [x] All [tests](https://github.com/jfrog/jfrog-cli-security/actions/workflows/test.yml) have passed. If this feature is not already covered by the tests, new tests have been added.
- [x] All changes are detailed at the description. if not already covered at [JFrog Documentation](https://github.com/jfrog/documentation), new documentation have been added.

-----

Audit is running multiple scans (Sca, Contextual analysis, secrets, IAC and SAST).
This PR adds the ability to control what type of sub scans will be preformed in the `audit` command. 

The following flags are added: `--sca`, `--secrets`, `--iac` and `--sast`, `--without-contextual-analysis`
* If none of those flags are specified - audit will run all the sub scans
* If one or more of those flags are specified, `audit` will only preform the specified sub scans
* `--without-contextual-analysis` can only be provided with `--sca` flag

Documentation changes: https://github.com/jfrog/documentation/pull/136

### CODE REVIEW

File: cli/docs/flags.go [hunk 1]
```diff
@@ -43,6 +43,13 @@ const (
 	Poetry = "poetry"
 )
 
+const (
```
Comment (hadarshjfrog) [inline cli/docs/flags.go:46]: why not using SubScanType?

File: cli/docs/flags.go [hunk 2]
```diff
@@ -219,6 +227,10 @@ var flagsMap = map[string]components.Flag{
 	),
 	RequirementsFile: components.NewStringFlag(RequirementsFile, "[Pip] Defines pip requirements file name. For example: 'requirements.txt'."),
 	CurationOutput:   components.NewStringFlag(OutputFormat, "Defines the output format of the command. Acceptable values are: table, json.", components.WithStrDefaultValue("table")),
+	Sca:              components.NewBoolFlag(Sca, "Set to true to request audit to only preform SCA sub scan."),
+	Iac:              components.NewBoolFlag(Iac, "Set to true to request audit to only preform IAC sub scan."),
+	Sast:             components.NewBoolFlag(Sast, "Set to true to request audit to only preform SAST sub scan."),
+	Secrets:          components.NewBoolFlag(Secrets, "Set to true to request audit to only preform Secrets sub scan."),
```
Comment (hadarshjfrog) [inline cli/docs/flags.go:233]: it'll be `--secrets=true` , or `--secrets` will be enough?

File: commands/audit/scarunner.go [hunk 3]
```diff
@@ -35,6 +36,10 @@ import (
 )
 
 func buildDepTreeAndRunScaScan(auditParallelRunner *utils.SecurityParallelRunner, auditParams *AuditParams, results *xrayutils.Results) (err error) {
+	if len(auditParams.ScansToPerform()) > 0 && !slices.Contains(auditParams.ScansToPerform(), xrayutils.ScaScan) {
+		log.Debug("Skipping SCA scan...")
```
Comment (hadarshjfrog) [inline commands/audit/scarunner.go:40]: Please add that it was because there was no dedicated flag

File: jas/runner/jasrunner.go [hunk 4]
```diff
@@ -35,14 +36,20 @@ func AddJasScannersTasks(securityParallelRunner *utils.SecurityParallelRunner, s
 	// Don't execute other scanners when scanning third party dependencies.
 	if !thirdPartyApplicabilityScan {
 		for _, module := range scanner.JFrogAppsConfig.Modules {
-			if err = addModuleJasScanTask(module, utils.Secrets, securityParallelRunner, runSecretsScan(securityParallelRunner, scanner, scanResults.ExtendedScanResults, module, secretsScanType), errHandlerFunc); err != nil {
+			if len(scansToPreform) > 0 && !slices.Contains(scansToPreform, utils.SecretsScan) {
```
Comment (hadarshjfrog) [inline jas/runner/jasrunner.go:39]: Please add that it was because there was no dedicated flag (for all scans)

File: utils/resultwriter.go [hunk 5]
```diff
@@ -122,24 +129,33 @@ func (rw *ResultsWriter) printScanResultsTables() (err error) {
...
+		if err = PrintIacTable(rw.results.ExtendedScanResults.IacScanResults, rw.results.ExtendedScanResults.EntitledForJas); err != nil {
+			return
+		}
 	}
-	if err = PrintIacTable(rw.results.ExtendedScanResults.IacScanResults, rw.results.ExtendedScanResults.EntitledForJas); err != nil {
-		return
+	if !slices.Contains(rw.subScansPreformed, SastScan) {
+		return nil
 	}
 	return PrintSastTable(rw.results.ExtendedScanResults.SastScanResults, rw.results.ExtendedScanResults.EntitledForJas)
```
Comment (hadarshjfrog) [inline utils/resultwriter.go:160]: 1. any reason to use ! Contains with nil, as oppose to the other that check "Contains" ?
2. BTW - why does the PrintSastTable is different that others? (doesn't return err)

File: cli/docs/flags.go (hunk 2 above)
Comment (attiasas) [reply cli/docs/flags.go:233]: both are valid

File: cli/docs/flags.go (hunk 1 above)
Comment (attiasas) [reply cli/docs/flags.go:46]: to seperate the CLI parsing from the logic

File: utils/resultwriter.go (hunk 5 above)
Comment (attiasas) [reply utils/resultwriter.go:160]: PrintSastTable does return err, its the end of the func...

File: cli/docs/flags.go [hunk 6]
```diff
@@ -43,6 +43,14 @@ const (
 	Poetry = "poetry"
 )
 
+const (
+	Sca                  = "sca"
+	Iac                  = "iac"
+	Sast                 = "sast"
+	Secrets              = "secrets"
+	NoContextualAnalysis = "without-contextual-analysis"
```
Comment (hadarshjfrog) [inline cli/docs/flags.go:51]: Internally I'd rename it to WithoutCA - just to keep it all a bit cleaner - but that's a personal preference - your call on that one.

File: cli/scancommands.go [hunk 7]
```diff
@@ -331,6 +331,30 @@ func AuditCmd(c *components.Context) error {
...
+	for _, subScan := range allSubScans {
+		if subScan == utils.ContextualAnalysisScan {
+			if c.GetBoolFlagValue(flags.NoContextualAnalysis) {
+				if !c.GetBoolFlagValue(flags.Sca) {
+					// No CA flag provided but sca flag is not provided as well, error
+					return pluginsCommon.PrintHelpAndReturnError(fmt.Sprintf("flag '--%s' cannot be used without '--%s'", flags.NoContextualAnalysis, flags.Sca), c)
+				}
+				continue
+			}
+		} else if !c.GetBoolFlagValue(subScan.String()) {
```
Comment (hadarshjfrog) [inline cli/scancommands.go:346]: meaning if a user will use just "--secrets" and not "--secrets=true" - it won't work?

Comment (attiasas) [reply cli/scancommands.go]: it will work both ways. `--secrets` == `--secrets=true` != `--secrets=false` or not exists

**Review by hadarshjfrog — APPROVED** [review]

**Pull request merged**